strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/providers/repository/event/RemoteRepositorySearchEvent.java

```diff
@@ -0,0 +1,22 @@
+package org.carlspring.strongbox.providers.repository.event;
+
+import org.carlspring.strongbox.event.Event;
+import org.carlspring.strongbox.providers.repository.RepositorySearchRequest;
+
+public class RemoteRepositorySearchEvent extends Event
+{
+
+    private RepositorySearchRequest eventData;
+
+    public RemoteRepositorySearchEvent(RepositorySearchRequest data)
+    {
+        super(-1);
```

Comment:
RepositoryEventTypeEnum ?

Reply:
Why -1? :)

Reply:
yeah, I understand that this looks strange :)
but we don't need to have additional `type` here like in `RepositoryEvent` or `ArtifactEvent`. `RemoteRepositorySearchEvent` typed by it's class and have only one purpose. So -1 need to pay attention to this fact, that we haven't forget to assign a type to it. With `-1` we assume that the type is just don't make sense here.

Reply:
I agree that event-api could be changed to be class-typed instead of enum-typed but this way will have two evolutions of this api and both approaches.

Reply:
I think that event can be typed using both `Class` and `eventId`, or just `Class`. There is no contradiction here.